### ISSUE TITLE
About training visual prompt

### ISSUE THREAD

**Issue by dyaxoio** [THU-MIG/yoloe#81]
When I fine-tuned the **Visual prompt** part, I found the dataloader still loads text feats to get **tpe**, then the **tpe** will get fused with **vpe** to obtain the **cls_pe**. Although the network of text part is frozen, I have thought the tpe would not get involved in the fine-tuning process of **Visual prompt** as we usually do not need to know the caption of objects inside **Visual prompt**.
Does the performance get worse when we only use vpe in the **Visual prompt** fine-tuning process? Or it is a typo in your code?

In [Line 730 of ultralytics/nn/tasks.py](https://github.com/THU-MIG/yoloe/blob/6e916ab2bca3da7cc304d98c6ff12167b768f757/ultralytics/nn/tasks.py#L730)
```python
elif isinstance(m, YOLOEDetect):
    vpe = m.get_vpe(x, vpe) if vpe is not None else None
    if return_vpe:
        assert(vpe is not None)
        assert(not self.training)
        return vpe
    cls_pe = self.get_cls_pe(m.get_tpe(tpe), vpe).to(device=x[0].device, dtype=x[0].dtype)
``` 
We can see:
```python
def get_cls_pe(self, tpe, vpe):
    all_pe = []
    if tpe is not None:
        assert(tpe.ndim == 3)
        all_pe.append(tpe)
    if vpe is not None:
        assert(vpe.ndim == 3)
        all_pe.append(vpe)
    if len(all_pe) == 0:
        all_pe.append(getattr(self, 'pe', torch.zeros(1, 80, 512)))
    return torch.cat(all_pe, dim=1)
```

**Comment by Cipheeer** [comment]
> 当我微调**视觉提示**部分时，我发现数据加载器仍然加载文本壮举来获得 **tpe**，然后 **tpe** 将与 **vpe** 融合以获得 **cls_pe**。虽然文本部分的网络被冻结了，但我认为 tpe 不会参与 **Visual Prompt** 的微调过程，因为我们通常不需要知道 **Visual Prompt** 中对象的标题。如果我们只在 **Visual 提示**符微调过程中使用 vpe，性能会变差吗？或者这是您的代码中的拼写错误？
> 
> 在 [ultralytics/nn/tasks.py 的第 730 行](https://github.com/THU-MIG/yoloe/blob/6e916ab2bca3da7cc304d98c6ff12167b768f757/ultralytics/nn/tasks.py#L730)
> 
> elif isinstance(m, YOLOEDetect):
>     vpe = m.get_vpe(x, vpe) if vpe is not None else None
>     if return_vpe:
>         assert(vpe is not None)
>         assert(not self.training)
>         return vpe
>     cls_pe = self.get_cls_pe(m.get_tpe(tpe), vpe).to(device=x[0].device, dtype=x[0].dtype)
> 我们可以看到：
> 
> def get_cls_pe(self, tpe, vpe):
>     all_pe = []
>     if tpe is not None:
>         assert(tpe.ndim == 3)
>         all_pe.append(tpe)
>     if vpe is not None:
>         assert(vpe.ndim == 3)
>         all_pe.append(vpe)
>     if len(all_pe) == 0:
>         all_pe.append(getattr(self, 'pe', torch.zeros(1, 80, 512)))
>     return torch.cat(all_pe, dim=1)

Hello, I would like to know how you approached fine-tuning the visual mechanism of the model. Here are my questions: I noticed that the introduction of VPE implies that the text network is frozen, which means the visual network is involved in the fine-tuning process. Is my understanding correct? Additionally, is your fine-tuning code based on the example notebook and does it retain SAVPE?
